tools/sanity_check/local_sanity_check.py.in

```diff
@@ -0,0 +1,269 @@
+                            "lsmcli vwcpu --vol %s --policy AUTO" % vol.id))
+
+            if flag_pass:
+                flag_has_pass = True
+
+        self._check_fail(fails)
+
+        if not flag_has_pass:
+            self._skip_current_test("No local disks are capable for this test")
+
+    def test_pool_status(self):
+        fails = []
+        flag_has_pass = False
+        pool_hash = {}
+        for pool in self.c.pools():
+            pool_hash[pool.id] = pool
+        pmi_hash = {}
+        disk_hash = {}
+        try:
+            for disk in self.c.disks():
+                disk_hash[disk.id] = disk
+        except LsmError as lsm_err:
+            if lsm_err.code == ErrorNumber.NO_SUPPORT:
+                pass
+            else:
+                raise
+
+        for vol_id in self.local_vol_hash.keys():
+            sys = self.local_vol_hash[vol_id]["sys"]
+            cap = self.local_vol_hash[vol_id]["cap"]
+            vol = self.local_vol_hash[vol_id]["vol"]
+            blk_paths = self.local_vol_hash[vol_id]["blk_paths"]
+            pool = pool_hash[vol.pool_id]
+            if pool.status == Pool.STATUS_UNKNOWN:
+                warn(vol, blk_paths,
+                     "Unknown pool status %d %s" % (pool.status,
+                                                    pool.status_info))
+            if not pool.status & Pool.STATUS_OK:
+                if supported(cap, [Cap.POOL_MEMBER_INFO]):
+                    # Check member failure
+                    if pool.id not in pmi_hash:
+                        pmi_hash[pool.id] = self.c.pool_member_info(pool)
+                    pmi = pmi_hash[pool.id]
+                    if pmi[1] == Pool.MEMBER_TYPE_DISK:
+                        for disk_id in pmi[2]:
+                            if not disk_hash[disk_id].status & Disk.STATUS_OK:
+                                fails.append(
+                                    Failure(blk_paths, vol, sys,
+                                            "Disk %s(%s) is not healthy: %d" %
+                                            (disk.name, disk_id, disk.status),
+                                            "Need investigation on output of"
+                                            "`lsmcli ld`"))
+                                continue
+
+                fails.append(
+                    Failure(blk_paths, vol, sys,
+                            "Pool is no healthy: %d '%s'" % (pool.status,
+                                                             pool.status_info),
+                            "Need investigation on output of `lsmcli lp`"))
+                continue
+            flag_has_pass = True
+
+        self._check_fail(fails)
+
+        if not flag_has_pass:
+            self._skip_current_test("No local disks are capable for this test")
+
+    def test_system_status(self):
+        fails = []
+        flag_has_pass = False
+        sys_hash = {}
+        for sys in self.c.systems():
```

Comment:
`setUp` already does some common data grabbing, perhaps you should extend it to have what you need here too?  Otherwise you are calling `self.c.systems()` multiple times.  At the very least you could use self.syss.